lib/solidus_dev_support/rspec/capybara.rb

```diff
@@ -9,12 +9,17 @@
 Capybara.default_max_wait_time = 10
 Capybara.server = :puma, { Silent: true } # A fix for rspec/rspec-rails#1897
 
-Capybara.drivers[:selenium_chrome_headless].tap do |original_driver|
-  Capybara.register_driver :solidus_chrome_headless do |app|
-    original_driver.call(app).tap do |driver|
-      driver.options[:options].args << "--window-size=#{CAPYBARA_WINDOW_SIZE.join(',')}"
-    end
-  end
+chrome_options = Selenium::WebDriver::Chrome::Options.new.tap do |options|
+  options.add_argument("--window-size=#{CAPYBARA_WINDOW_SIZE.join(',')}")
+  options.add_argument("--headless")
+  options.add_argument("--disable-gpu")
+end
+
+version = Capybara::Selenium::Driver.load_selenium
+options_key = Capybara::Selenium::Driver::CAPS_VERSION.satisfied_by?(version) ? :capabilities : :options
+
+Capybara.register_driver :solidus_chrome_headless do |app|
+  Capybara::Selenium::Driver.new(app, browser: :chrome, options_key => chrome_options)
 end
```

Comment:
Instead of constructing the `solidus_chrome_headless` driver from scratch, have you considered calling [Capybara::Driver#resize_window_to](https://www.rubydoc.info/gems/capybara/Capybara/Selenium/Driver#resize_window_to-instance_method) to resize the `solidus_chrome_headless` window? I was able to get the following change to pass the SolidusAuthDevise specs:

```rb
CAPYBARA_WINDOW_SIZE = (ENV['CAPYBARA_WINDOW_SIZE'] || '1920x1080').split('x', 2).map(&:to_i)
CAPYBARA_WINDOW_WIDTH = CAPYBARA_WINDOW_SIZE[0]
CAPYBARA_WINDOW_HEIGHT = CAPYBARA_WINDOW_SIZE[1]

# ...

Capybara.drivers[:selenium_chrome_headless].tap do |original_driver|
  Capybara.register_driver :solidus_chrome_headless do |app|
    original_driver.call(app).tap do |driver|
      # Original code: driver.options[:options].args << "--window-size=#{CAPYBARA_WINDOW_SIZE.join(',')}"

      driver.resize_window_to(driver.current_window_handle, CAPYBARA_WINDOW_WIDTH, CAPYBARA_WINDOW_HEIGHT)
    end
  end
end
```